devdoc/Metrics.rst

```diff
@@ -0,0 +1,212 @@
+=================
+Metrics Framework
+=================
+
+************
+Introduction
+************
+
+The metrics framework provides a set of classes and templates for implementing a metrics collection strategy
+in the platform.
+
+The framework consists of three main areas. Each is responsible for a portion of the collection and reporting
+operations. These areas are
+
+* Metrics
+* Sinks
+* The Metrics Reporter
+
+*******
+Metrics
+*******
+
+Metrics represent the base value that a component must track. The framework defines a number of metric
+classes for use by components. Each metric class encapsulates all of the necessary logic to handle the
+following:
+
+* Updating the value being tracked
+* Synchronization between updating and collecting
+* Providing measurements to the framework
+
+Components are only responsible for creating metric objects, instrumenting the component to update
```

Comment:
The intent is for components to use framework provided classes as they provide all of the necessary interfaces for the reporter and handle synchronization. If there are other metrics needed, as part of the process of instrumenting each component we can add new metric classes. Or, components can provide their own subclass to the metric base class.

Reply:
Since you haven't worked through any real world examples of metrics I would say it is too early to decide that all metrics will be gathered by metric classes.  E.g. If the metric is already being gathered then does that mean it now needs to be gathered twice?  You could refactor so that it is collected by the new classes, but what happens if that doesn't fit in with how it is colleted?